drivers/flash/nrf_qspi_nor.c

```diff
@@ -133,13 +133,28 @@ BUILD_ASSERT(DT_INST_PROP(0, cpol) == DT_INST_PROP(0, cpha),
 #define QSPI_PROP_AT(prop, idx) DT_PROP_BY_IDX(QSPI_NODE, prop, idx)
 #define QSPI_PROP_LEN(prop) DT_PROP_LEN(QSPI_NODE, prop)
 
-#define INST_0_QER _CONCAT(JESD216_DW15_QER_, \
+#define INST_0_QER _CONCAT(JESD216_DW15_QER_VAL_, \
 			   DT_STRING_TOKEN(DT_DRV_INST(0), \
 					   quad_enable_requirements))
 
-BUILD_ASSERT(((INST_0_QER == JESD216_DW15_QER_NONE)
-	      || (INST_0_QER == JESD216_DW15_QER_S1B6)),
-	     "Driver only supports NONE or S1B6 for quad-enable-requirements");
+#define SR1_WRITE_CLEARS_SR2 (INST_0_QER == JESD216_DW15_QER_VAL_S2B1v1)
```

Comment:
Sorry but this will not pass this check: https://github.com/zephyrproject-rtos/zephyr/blob/main/scripts/checkpatch.pl#L4980 because this is still covered by second group in regex.
You would have to do extra thing like:
```
#define IS_EQUAL(x, y) ((x) == (y))
```